src/mds/Server.cc

```diff
@@ -463,8 +463,13 @@ void Server::reclaim_session(Session *session, const cref_t<MClientReclaim> &m)
     finish_reclaim_session(session, reply);
     return;
   }
-
-  ceph_abort();
+  if(m->get_connection()->get_peer_type() == CEPH_ENTITY_TYPE_CLIENT) {
+    derr << "Server received unknown message " << m->get_type() << ", closing session and blocklisting the client " << session->get_client() << dendl;
+    CachedStackStringStream css;
+    mds->evict_client(session->get_client().v, false, true, *css, nullptr);
+  } else {
+      ceph_abort_msg("Aborting: server unknown message  " + to_string(m->get_type()) + " from peer type " + to_string(m->get_connection()->get_peer_type()));
```

Comment:
This seems unnecessary, the reclaim requests only come from client side. As @batrick mentioned in previous comments for another place.

Here you only need to check whether there has any unknown reclaim flag bit set, if so just evict the clients.

Reply:
If it doesn't return from the the if statement which checks for `flags & CEPH_RECLAIM_RESET` then it indicates there is some unknown bit, so client can be directly evicted, but just to make sure it's the client we are evicting, don't you think it would be good to make sure it's the client before we evict it?